test/remote/gateways/remote_worldpay_test.rb

```diff
@@ -456,6 +457,27 @@ def test_marking_3ds_purchase_as_moto
     assert_equal 'SUCCESS', response.message
   end
 
+  def test_successful_authorize_with_3ds2_challenge
```

Comment:
Does this test case cover the scenario of multiple requests?

Reply:
Good question, I was planning to add a more complete test for that to the Core side since it's easier to simulate the multiple requests. I'm not sure how to simulate those in AM because some of that is handled on the Core side, and I couldn't find any examples of a complete challenge flow in AM tests 😅

Reply:
Gotcha, I'm a little lost on the use of `@cookie`. I didn't realize things persisted in AM. Does using `@cookie` mean that all subsequent calls to AM for a given transactions get access to it?

Reply:
I have the same question ^^

Reply:
No, you're right, there will need to be a corresponding Core change that saves each cookie. After some more testing/docs peering, I'll need to change this to reset the cookie each time a new cookie is sent back and parse out part of the string before saving. This is more complicated than I thought at first 😓  Will push some updates later today!

Reply:
updated the top comment with some helpful context!